src/_pytest/fixtures.py

```diff
@@ -1697,11 +1722,16 @@ def parsefactories(  # noqa: F811
     def getfixturedefs(
         self, argname: str, nodeid: str
     ) -> Optional[Sequence[FixtureDef[Any]]]:
```

Comment:
Is `getfixturedefs` public API? If not, we could (in a future PR of course) change the return value to make the subtle difference between `None` and empty list more explicit.

Reply:
It's not public. I'm not sure how to make it better though, interested to hear what you had in mind.

Reply:
I had not thought specifics when I left the comment, but perhaps something to make it more type-safe and avoid chances for misuse:

```python
@dataclass(frozen=True)
class UndefinedFixture:
    pass

@dataclass(frozen=True)
class NoFixturesApplicable:
    pass

def getfixturedefs(
    self, argname: str, nodeid: str
) -> Sequence[FixtureDef[Any]] | NoFixturesApplicable | UndefinedFixture:
```

This would force the caller to handle each individual case...

However perhaps this is sufficient:

```python
@dataclass(frozen=True)
class UndefinedFixture:
    pass

def getfixturedefs(
    self, argname: str, nodeid: str
) -> Sequence[FixtureDef[Any]] | UndefinedFixture:
```

The original code feels like a small smell because we need to document the difference between `None` and `[]`, and it is easy to treat both equally in a bool context.

But not so sure if worth it or not.